mlflow/store/artifact/models_artifact_repo.py

```diff
@@ -140,16 +144,22 @@ def list_artifacts(self, path):
         return self.repo.list_artifacts(path)
 
     def _add_registered_model_meta_file(self, model_path):
-        write_yaml(
-            model_path,
-            REGISTERED_MODEL_META_FILE_NAME,
-            {
-                "model_name": self.model_name,
-                "model_version": self.model_version,
-            },
-            overwrite=True,
-            ensure_yaml_extension=False,
-        )
+        if os.path.isdir(model_path):
+            write_yaml(
+                model_path,
+                REGISTERED_MODEL_META_FILE_NAME,
+                {
+                    "model_name": self.model_name,
+                    "model_version": self.model_version,
+                },
+                overwrite=True,
+                ensure_yaml_extension=False,
+            )
+        else:
+            _logger.warning(
+                "Registered Model Metadata file not able to be written. Destination path "
+                f"'{model_path}' is not a directory."
+            )
```

Comment:
This warning might scare users.

Reply:
Can we remove the else clause?

Reply:
Agreed. Removed this. It's really only valid for deploying a root model directory and only if deploying to MLserver anyway. I'll add a note in the code about why the file might not be created.